data/src/error_analysis/README.md

```diff
@@ -0,0 +1,45 @@
+# Error Analysis tool for G2P
+
+This tool could be used for performing fine-grain error analysis of a G2P model. It reports a performance matrix of a test output generated by a G2P model. 
+
+Under this error analysis process, each test record generated from a G2P model is evaluated under two criterions. 
+a) if the hypothesized form matches the corpus prediction or not.
+b) if the hypothesized form adheres to grammatical rules of the language.
+
+The output of the tool is a two dimentional matrix as shown below-
+
+```
+               | CG Match  |   CG Not Match |
+---------------|-----------+----------------|
+Pron Match     |  0.7934   |      0.0501    |
+Pron Not Match |  0.1315   |      0.0250    |
+```
+
+Here the numbers represents percentage of test records falling under X section of each category. 
+
+## Prerequisite
+
+### Library
+install pynini 
+
+```
+pip install pynini
+```
+
+### Data
+To run this tool following data items are required -
+
+a) Covering Grammer : Each line contain graphime and their corresponding pronunciation seperated by a tab. Refer [cg.tsv](data/cg.tsv) file inside data folder for more detail.
+
+b) Test output : This file contains three attributes in each line seperated by a tab. The attributes are - orthography, expected pronunciation, and hypothesized pronunciation (placed in the same order). Refer [test.tsv](data/test.tsv) file inside data folder for more detail.
+
+
+## How to Run
+clone repo
+```
+cd g2pErrorAnalysis
+python erroranalysis.py --cg_path data/cg.tsv --test_path data/test.tsv
+```
+
+## Reference:
+Gorman, Kyle. 2016. Pynini: a Python library for weighted finite-state gram- mar compilation. In Proceedings of the SIGFSM workshop on statistical NLP and weighted automata, 75–80. Berlin: Association for Computational Lin- guistics. https://www.aclweb.org/anthology/W16-2409.
```

Comment:
typo: `Lin- guististics`. Instead of including the URL in the citation, you could make the title a link to the URL: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links

Reply:
fixed.